docs_user/modules/proc_migrating-the-rgw-backends.adoc

```diff
@@ -0,0 +1,172 @@
+[id="migrating-the-rgw-backends_{context}"]
+
+= Migrating the {Ceph} RGW backends
+
+To match the cardinality diagram, you use cephadm labels to refer to a group of nodes where a given daemon type should be deployed. For more information about the cardinality diagram, see xref:ceph-daemon-cardinality_{context}[{Ceph} daemon cardinality].
+
+//kgilliga: Should each instance of "CephStorage" node be changed to "Red Hat Ceph Storage node" (downstream) and "Ceph Storage node" (upstream)?
+
+.Procedure
+
+. Add the RGW label to the {Ceph} nodes:
++
+----
+for i in 0 1 2; {
+    ceph orch host label add cephstorage-$i rgw;
+}
+----
++
+----
+[ceph: root@controller-0 /]#
+
+for i in 0 1 2; {
+    ceph orch host label add cephstorage-$i rgw;
+}
+
+Added label rgw to host cephstorage-0
+Added label rgw to host cephstorage-1
+Added label rgw to host cephstorage-2
+
+[ceph: root@controller-0 /]# ceph orch host ls
+
+HOST       	ADDR       	LABELS      	STATUS
+cephstorage-0  192.168.24.54  osd rgw
+cephstorage-1  192.168.24.44  osd rgw
+cephstorage-2  192.168.24.30  osd rgw
+controller-0   192.168.24.45  _admin mon mgr
+controller-1   192.168.24.11  _admin mon mgr
+controller-2   192.168.24.38  _admin mon mgr
+
+6 hosts in cluster
+----
+
+. During the overcloud deployment, RGW is applied at step2
+(external_deployment_steps), and a cephadm compatible spec is generated in
+`/home/ceph-admin/specs/rgw` from the https://github.com/openstack/tripleo-ansible/blob/master/tripleo_ansible/ansible_plugins/modules/ceph_mkspec.py[ceph_mkspec] ansible module.
+Find and patch the RGW spec, specifying the right placement using the labels
+approach, and change the rgw backend port to 8090 to avoid conflicts
+with the https://github.com/openstack/tripleo-ansible/blob/master/tripleo_ansible/roles/tripleo_cephadm/tasks/rgw.yaml#L26-L30[Ceph Ingress Daemon] (*)
+//kgilliga: This step needs to be rewritten. We shouldn't link to github in the downstream docs. And are the "external_deployment_steps" pending?
```

Comment:
I agree, we can rewrite this part and make it better.
The idea is:
- take the spec (L54 to L67)
- modify the placement section and put `label: rgw`: this results in the new spec

```

networks:
- 172.17.3.0/24
placement:
  label: rgw
service_id: rgw
service_name: rgw.rgw
service_type: rgw
spec:
  rgw_frontend_port: 8090
  rgw_realm: default
  rgw_zone: default
```
Note: change the port from `8080` to `8090` to avoid conflicts with the rgw instance.

Apply the new RGW spec using the orchestrator CLI:

```
$ cephadm shell -m /home/ceph-admin/specs/rgw -- ceph orch apply -i /mnt/rgw
```